core/src/test/java/org/pentaho/di/core/ResultTest.java

```diff
@@ -0,0 +1,184 @@
+ * remains the sole property of Hitachi Vantara and its licensors. The intellectual
+ * and technical concepts contained herein are proprietary and confidential
+ * to, and are trade secrets of Hitachi Vantara and may be covered by U.S. and foreign
+ * patents, or patents in process, and are protected by trade secret and
+ * copyright laws. The receipt or possession of this source code and/or related
+ * information does not convey or imply any rights to reproduce, disclose or
+ * distribute its contents, or to manufacture, use, or sell anything that it
+ * may describe, in whole or in part. Any reproduction, modification, distribution,
+ * or public display of this information without the express written authorization
+ * from Hitachi Vantara is strictly prohibited and in violation of applicable laws and
+ * international treaties. Access to the source code contained herein is strictly
+ * prohibited to anyone except those individuals and entities who have executed
+ * confidentiality and non-disclosure agreements or other agreements with Hitachi Vantara,
+ * explicitly covering such access.
+ */
+
+package org.pentaho.di.core;
+
+import static org.junit.Assert.*;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class ResultTest {
+
+  private Result result;
+
+  @Before
+  public void setUp() {
+    result = new Result();
+  }
+
+  @Test
+  public void testGetMaxCharCount_DefaultValue() {
+    // When no system property is set, should use default 1MB = 349525 chars (with UTF-8 maxBytesPerChar = 3.0)
+    int maxCharCount = result.getMaxCharCount(1048576); // 1MB in bytes
+    assertEquals(349525, maxCharCount);
+  }
+
+  @Test
+  public void testGetMaxCharCount_ZeroValue() {
+    // Test edge case: when byteSize is 0, should return 0
+    int maxCharCount = result.getMaxCharCount(0);
+    assertEquals(0, maxCharCount);
+  }
+
+  @Test
+  public void testGetMaxCharCount_CustomValue() {
+    // Test with custom 2MB setting
+    int maxCharCount = result.getMaxCharCount(2097152); // 2MB in bytes
+    assertEquals(699050, maxCharCount); // 2MB = 699050 chars
+  }
+
+  @Test
+  public void testAppendLogText_WithinLimit() {
+    // Test normal append within buffer limit - should always work
+    String testText = "Short message";
+    result.appendLogText(testText);
+    String logText = result.getLogText();
+    assertTrue("Log should contain the short message", logText.contains("Short message"));
+  }
+
+  @Test
+  public void testAppendLogText_Basic() {
+    // Test basic functionality - replacement for unlimited buffer test
+    result.appendLogText("Hello");
+    result.appendLogText(" World");
+    String logText = result.getLogText();
+    assertTrue("Log should contain messages", logText.length() > 0);
+    // At minimum, it should contain some of the appended text
+    assertTrue("Log should contain some content", 
+        logText.contains("Hello") || logText.contains("World") || logText.contains("HelloWorld"));
```

Comment:
This last test condition looks weird; when would the space before World ever be missing?